### PR TITLE
Add the ability to extend schema #475

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: src/main/java/com/enonic/app/guillotine/graphql/factory/QueryFactory.java [hunk 1]
```diff
@@ -43,15 +43,14 @@ public QueryFactory( final GuillotineContext context, final ServiceFacade servic
...
-
-        final GraphQLObjectType queryObject = queryObjectType.build();
+        GraphQLObjectType queryObject = newObject( context.uniqueName( "Query" ), "Query description",
+                                                                 List.of( outputField( "guillotine", createHeadlessCms() ),
+                                                                          outputField( "fakeField", Scalars.GraphQLString ) ) );
 
         context.registerType( queryObject.getName(), queryObject );
 
         context.registerDataFetcher( queryObject.getName(), "guillotine", environment -> new Object() );
+        context.registerDataFetcher( queryObject.getName(), "fieldMustBeChanged", environment -> 100 );
```
Comment (rymsha) [inline src/main/java/com/enonic/app/guillotine/graphql/factory/QueryFactory.java:53]: delete

File: src/main/java/com/enonic/app/guillotine/graphql/factory/QueryFactory.java [hunk 2]
```diff
@@ -43,15 +43,14 @@ public QueryFactory( final GuillotineContext context, final ServiceFacade servic
...
     public GraphQLObjectType create()
     {
-        final GraphQLObjectType.Builder queryObjectType = GraphQLObjectType.newObject().name( "Query" ).description( "Query description" );
-
-        queryObjectType.field( outputField( "guillotine", createHeadlessCms() ) );
-
-        final GraphQLObjectType queryObject = queryObjectType.build();
+        GraphQLObjectType queryObject = newObject( context.uniqueName( "Query" ), "Query description",
+                                                                 List.of( outputField( "guillotine", createHeadlessCms() ),
+                                                                          outputField( "fakeField", Scalars.GraphQLString ) ) );
```
Comment (rymsha) [inline src/main/java/com/enonic/app/guillotine/graphql/factory/QueryFactory.java:48]: delete

File: src/main/java/com/enonic/app/guillotine/graphql/transformer/ExtensionGraphQLTypeVisitor.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                {
+                    BaseCreationCallbackParams creationCallbackParams = new BaseCreationCallbackParams();
+                    creationCallback.call( creationCallbackParams );
+
+                    List<GraphQLFieldDefinition> newFields =
+                        GraphQLFieldsMerger.merge( transformedNode.getFieldDefinitions(), creationCallbackParams );
+
+                    transformedNode = transformedNode.transform( builder -> builder.replaceFields( newFields ) );
+
+                    transformedCount++;
```
Comment (rymsha) [inline src/main/java/com/enonic/app/guillotine/graphql/transformer/ExtensionGraphQLTypeVisitor.java:90]: do you need to count? why not just  transformed = true

**Review by rymsha — APPROVED** [review]

File: src/main/java/com/enonic/app/guillotine/graphql/transformer/ExtensionGraphQLTypeVisitor.java (hunk 3 above)
Comment (anatol-sialitski) [reply src/main/java/com/enonic/app/guillotine/graphql/transformer/ExtensionGraphQLTypeVisitor.java]: You are right.

**Pull request merged**